generator/booking_scoring.cpp

```diff
@@ -0,0 +1,97 @@
+#include "generator/booking_scoring.hpp"
+
+#include "generator/booking_dataset.hpp"
+
+#include "indexer/search_string_utils.hpp"
+#include "indexer/search_delimiters.hpp"
+
+#include "geometry/distance_on_sphere.hpp"
+
+#include "base/collection_cast.hpp"
+
+namespace generator
+{
+namespace booking_scoring
+{
+namespace
+{
+// Calculated with tools/python/booking_hotels_quality.py.
+double constexpr kOptimalThreshold = 0.151001;
+
+template <typename T, typename U>
+struct decay_equiv :
+    std::is_same<typename std::decay<T>::type, U>::type
+{};
+
+set<strings::UniString> StringToSetOfWords(string const & str)
+{
+  vector<strings::UniString> result;
+  search::NormalizeAndTokenizeString(str, result, search::Delimiters{});
+  return my::collection_cast<set>(result);
```

Comment:
Do you really need a set here? As you do not modify the vector, why not to just sort it before return, and get rid of set?